internal/kafka/kafka_rest_onprem.go

```diff
@@ -12,21 +12,31 @@ import (
 
 	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
 	"github.com/confluentinc/cli/v3/pkg/errors"
+	"github.com/confluentinc/cli/v3/pkg/kafkarest"
 	"github.com/confluentinc/cli/v3/pkg/output"
 )
 
-func initKafkaRest(c *pcmd.AuthenticatedCLICommand, cmd *cobra.Command) (*kafkarestv3.APIClient, context.Context, error) {
+func initKafkaRest(c *pcmd.AuthenticatedCLICommand, cmd *cobra.Command) (*kafkarestv3.APIClient, context.Context, string, error) {
 	url, err := getKafkaRestUrl(cmd)
 	if err != nil { // require the flag
-		return nil, nil, err
+		return nil, nil, "", err
 	}
 	kafkaREST, err := c.GetKafkaREST()
```

Comment:
I think we might be able to unify everything under the `c.GetKafkaREST()` function? 

We can call the `kafka_rest_onprem` helpers to resolve the server URL with `c.command` before returning the `KafkaREST` struct in `kafka_rest.go` if we're on-prem. Finally, we can just add an on-prem case for getting the cluster ID. This way everything can be accessed from the same struct whether we're on-prem or cloud.

Reply:
Yup, that's the goal. This is just to clean it up a bit in the meantime.